sklearn/compose/_target.py

```diff
@@ -175,8 +175,7 @@ def fit(self, X, y, **fit_params):
         -------
         self : object
         """
-        y = check_array(y, accept_sparse=False, force_all_finite=True,
-                        ensure_2d=False, dtype='numeric')
+        y = np.asarray(y)
```

Comment:
I am not sure this is the right things to do. Now passing a sparse matrix will give something funky and will error, while the `check_array` was used to filter this part (in addition of checking for non-finite values).

I think the safe thing to do here is to allow for nd array using `allow_nd=True` in `check_array`.